src/lib/falcor/FalcorRouter.js

```diff
@@ -11,25 +28,169 @@ export default class FalcorRouter extends BaseRouter.createClass([
+        });
+        query += "&fields=slug"
+        pathSet[2].forEach((field, index) => {
+          if (field !== 'slug') {
+            query += "," + ghostify(field);
+          }
+        })
+        query += "&limit=" + pathSet.slugs.length.toString();
+        ghostArticleQuery(query).then((data) => {
+          data = data.posts;
+          const results = [];
+          data.forEach((article, index) => {
+            pathSet[2].forEach((field) => {
+              let ghostField = ghostify(field);
+              results.push({
+                path: ["articlesBySlug", article.slug, field],
+                value: article[ghostField]
+              });
+            });
+          });
+          resolve(results);
+        });
+      });
+    }
+  },
+  {
+    //Get custom article data from SQL database
+    route: "articlesBySlug[{keys:slugs}]['issue', 'category', 'description']",
+    get: (pathSet) => {
+      return new Promise((resolve, reject) => {
+        let query = {slug: {$in: pathSet.slugs}};
+        let projection = {_id: 0};
+        pathSet[2].forEach((field) => {
+          projection[field] = 1;
+        });
+        projection['slug'] = 1;
+        SQLArticleQuery(query, projection).then((data) => {
+          const results = [];
+          data.forEach((article) => {
+            pathSet[2].forEach((field) => {
+              results.push({
+                path: ["articlesBySlug", article.slug, field],
+                value: article[field]
+              });
+            });
+          });
+          resolve(results);
         });
+      });
+    }
+  },
+  {
+    // Get author information from article
+    route: "articlesBySlug[{keys:slugs}]['authors'][{integers:indices}]",
+    get: (pathSet) => {
+      console.log("here");
+      return new Promise((resolve, reject) => {
+        console.log("there");
+        let query = {slug: {$in: pathSet.slugs}};
+        let projection = {_id: 0, slug: 1, authors: 1};
+        SQLArticleQuery(query, projection).then((data) => {
+          console.log("made it and data coming here");
+          const results = [];
+          console.log(data);
+          data.forEach((article) => {
+            let authors = article.authors;
+            let indices = pathSet.indices;
+            for (let i = 0; i < indices.length; i++) {
+              let index = indices[i];
+              if (index >= authors.length) {
+                continue;
+              }
+              results.push({
+                path: ['articlesBySlug', article.slug, 'authors', index],
+                value: $ref(['authorsBySlug', authors[index]])
+              });
+            }
+          });
+          console.log("resolving");
+          console.log(results);
+          resolve(results);
+        });
+      });
+    }
+  },
+  {
+    // Get articles in chronological order by a consecutive range of indices where both min and max are multiples
```

Comment:
Yes I would avoid Atoms

On Sat, Jul 16, 2016, 1:51 AM Emil Goldsmith Olesen <
notifications@github.com> wrote:

> In src/lib/falcor/FalcorRouter.js
> <https://github.com/thegazelle-ad/gazelle-front-end/pull/9#discussion_r71051009>
> :
>
> > +              }
> > +              results.push({
> > +                path: ['articlesBySlug', article.slug, 'authors', index],
> > +                value: $ref(['authorsBySlug', authors[index]])
> > +              });
> > +            }
> > +          });
> > +          console.log("resolving");
> > +          console.log(results);
> > +          resolve(results);
> > +        });
> > +      });
> > +    }
> > +  },
> > +  {
> > +    // Get articles in chronological order by a consecutive range of indices where both min and max are multiples
>
> Oh yeah of course, I see what you mean.
>
> Yeah in that case the new #3
> <https://github.com/thegazelle-ad/gazelle-front-end/issues/3> might be
> very ugly, but is clear and will work.
>
> And it's better style to do it with the last part where we call
> [articleOnPage] than returning an atom with the page right?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/thegazelle-ad/gazelle-front-end/pull/9/files/cd3df8271a1301dd380cbce86fc65f61794aa5fd#r71051009>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADAFzYfUm2uKlSBOI8Du8eHn5szm7ij9ks5qWA7YgaJpZM4JMWcC>
> .
>